docs/migration.md

```diff
@@ -1,61 +1,79 @@
+!!! note
+    Since v2.0.0, GPUStack supports Linux only, and this migration must be run on Linux.
 
-Find the path to your existing data directory used by the legacy installation. The default path is:
+### Step 1: Identify Your Legacy Data Directory
+
+Locate the data directory used by your previous GPUStack installation. The default path is:
 
 ```bash
 /var/lib/gpustack
 ```
 
-We'll refer to this as `${your-data-dir}` in the next step.
+In the following steps, this path is referenced as `${your-data-dir}`.
+
+### Step 2: Migrate Using Docker
 
-### Step 2: Reinstall GPUStack via Docker
+#### Server Migration (NVIDIA GPUs)
 
-If you are using Nvidia GPUs, run the following Docker command to migrate your GPUStack server, replacing the volume mount path with your data directory location.
+If you are using Nvidia GPUs, run the following Docker command to start the migration. Replace `${your-data-dir}` with the path to your legacy data directory, which must contain the original SQLite database and related files. 
+
+By mounting `${your-data-dir}` to `/var/lib/gpustack` and setting the environment variable `GPUSTACK_MIGRATION_DATA_DIR`, GPUStack will automatically detect and migrate your SQLite data to the latest embedded PostgreSQL database during startup.
 
 ```bash
-docker run -d --name gpustack \
-    --restart=unless-stopped \
-    --gpus all \
-    --network=host \
-    --ipc=host \
-    -v ${your-data-dir}:/var/lib/gpustack \
-    gpustack/gpustack
+sudo docker run -d --name gpustack-server \
+      --restart=unless-stopped \
+      --privileged \
+      --network=host \
+      --env GPUSTACK_MIGRATION_DATA_DIR=/var/lib/gpustack \
+      --volume /var/run/docker.sock:/var/run/docker.sock \
+      --volume ${your-data-dir}:/var/lib/gpustack \
+      --runtime nvidia \
+      gpustack/gpustack
 ```
 
-This will launch GPUStack using Docker, preserving your existing data.
+This command will launch the GPUStack server in Docker, preserving and migrating your existing data.
+
+#### Worker Migration (NVIDIA GPUs)
+
+For worker nodes, replace `${your-data-dir}` with the legacy worker data directory path. Use the following command:
 
-For workers and other GPU hardware platforms, please refer to the commands in the [Installation Documentation](installation/installation-requirements.md).
+```bash
+sudo docker run -d --name gpustack-worker \
+      --restart=unless-stopped \
+      --privileged \
+      --network=host \
+      --volume /var/run/docker.sock:/var/run/docker.sock \
+      --volume ${your-data-dir}:/var/lib/gpustack \
+      --runtime nvidia \
+      gpustack/gpustack \
+      --server-url ${server-url} \
+      --token ${token}
+```
 
-## macOS / Windows Migration
+This will launch the GPUStack worker using your existing data and connect it to the specified server.
 
-Download and install the new version of GPUStack via [Desktop Installer](./installation/desktop-installer.md#download-installer).
+#### Other GPU Architectures
```

Comment:
Installation update is merged, we can tell users to refer to the related docs for these cases.